tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleModuleProjectDeleteParticipant.java

```diff
@@ -0,0 +1,223 @@
+			return new RefactoringStatus();
+		}
+
+		public abstract Change perform(IProgressMonitor monitor) throws CoreException;
+
+		protected IProject deleteProject;
+
+	}
+
+	public class RmoveModulePostChange extends RemoveModuleChange {
+
+		public RmoveModulePostChange(IProject project) {
+			super(project);
+		}
+
+		@Override
+		public Change perform(IProgressMonitor monitor) throws CoreException {
+			if (_liferayWorkspaceProject == null) {
+				return null;
+			}
+
+			if (ListUtil.isNotEmpty(_projectsToWatch)) {
+				_liferayWorkspaceProject.watch(_projectsToWatch);
+			}
+
+			return null;
+		}
+
+	}
+
+	public class RmoveModulePreChange extends RemoveModuleChange {
+
+		public RmoveModulePreChange(IProject project) {
+			super(project);
+		}
+
+		@Override
+		public Change perform(IProgressMonitor monitor) throws CoreException {
+			if (_liferayWorkspaceProject == null) {
+				return null;
+			}
+
+			IProject workspaceProject = _liferayWorkspaceProject.getProject();
+
+			if (!_projectsToWatch.contains(workspaceProject)) {
+				Stream<IProject> watchStream = _projectsToWatch.stream();
+
+				_projectsToWatch = watchStream.filter(
+					project -> !project.equals(deleteProject)
+				).collect(
+					Collectors.toSet()
+				);
+			}
+			else if (deleteProject.equals(workspaceProject)) {
+				_projectsToWatch.remove(deleteProject);
+			}
+
+			String jobName = workspaceProject.getName() + ":watch";
+
+			IJobManager jobManager = Job.getJobManager();
+
+			Job[] jobs = jobManager.find(jobName);
+
+			if (ListUtil.isNotEmpty(jobs)) {
+				Job job = jobs[0];
+
+				job.cancel();
+
+				try {
+					job.join();
+				}
+				catch (InterruptedException ie) {
+				}
+			}
+
+			return null;
+		}
+
+	}
+
+	@Override
+	protected boolean initialize(Object element) {
+		if (!(element instanceof IProject)) {
+			return false;
+		}
+
+		DeleteArguments arguments = getArguments();
+
+		if (!arguments.getDeleteProjectContents()) {
+			return false;
+		}
+
+		_deleteProject = (IProject)element;
+
+		try {
+			boolean gradleProject = GradleUtil.isGradleProject(_deleteProject);
+
+			if (!gradleProject) {
+				return false;
+			}
+
+			IProject workspaceProject = LiferayWorkspaceUtil.getWorkspaceProject();
+
+			_liferayWorkspaceProject = LiferayCore.create(IWorkspaceProject.class, workspaceProject);
+
+			if (_liferayWorkspaceProject != null) {
+				_projectsToWatch = new HashSet<>(_liferayWorkspaceProject.watching());
+
+				_childProjects = new HashSet<>(_liferayWorkspaceProject.getChildProjects());
+
+				if (ListUtil.contains(_projectsToWatch, workspaceProject)) {
+					if (_deleteProject.equals(workspaceProject)) {
+						return true;
+					}
+
+					return _childProjects.contains(_deleteProject);
+				}
+
+				return _projectsToWatch.contains(_deleteProject);
+			}
+		}
+		catch (Exception e) {
+		}
+
+		return false;
+	}
+
+	private Set<IProject> _childProjects;
```

Comment:
You don't need all of these 4 variables.  

All you need are the following:
1. save off the project to be deleted.
2. save off the list of projects that should be watched after the  project has been deleted.